### PR TITLE
Mpeg:Improves the support of multiple streams video.

### LINKED ISSUE #8526: Xyanide Resurrection freezing
Seems like a scempeg bug

log
https://gist.github.com/daniel229/15538499b882fcf3a39c

### PR DESCRIPTION
**Pull request by shenweip**
Fixes #8526. 

That game has strict requirements for the number of video frames. If no enough packets are read, the game will hang. 

And the videos it uses have 4 video streams, but the mpegheader seems don't have the correct stream info.

### CODE REVIEW

**Comment by hrydgard** [comment]
Code looks good. Let's try it.

**Comment by Panderner** [comment]
this will broke Jeanne d'Arc and Open Season unfortunately.

**Comment by shenweip** [comment]
Don't worry, I know how to fix now:)

File: Core/HLE/sceMpeg.cpp [hunk 1]
```diff
@@ -1495,6 +1496,9 @@ void PostPutAction::run(MipsCall &call) {
 // Program signals that it has written data to the ringbuffer and gets a callback ?
 static u32 sceMpegRingbufferPut(u32 ringbufferAddr, int numPackets, int available)
 {
+	// Generally, program will call sceMpegRingbufferAvailableSize() before this func.
+	// Still need to check available?
```
Comment (unknownbrackets) [inline Core/HLE/sceMpeg.cpp:1500]: See #978.  I think we should test this, though, for the correct behavior.  Ideally we should replicate (as far as is visible) what firmware does.

-[Unknown]

**Pull request merged**